### PR TITLE
Add tests to taskstatus

### LINKED ISSUE #564: Create unit tests in some class's methods
Hi, @flyve-mdm/backend-development.

**Is your feature request related to a problem? Please describe.**
Currently, in the code-coverage report there is missing unit test in some class or class method.
**Describe the solution you'd like**
Create the test methods corresponding to everyone missing.

This is the coverage report:

image

### LINKED ISSUE #676: Add more unit tests for release 2.0
- [x] Task status tests

### PR DESCRIPTION
**Pull request by ingluife**
Hi, @flyve-mdm/backend-development.

### Changes description
I add the test indicated in the code-coverage report that missing in the Unit test.
The changes are:

- add testUpdateStatus method
- add testPrepareInputForUpdate method

### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@ingluife |4|
|@btry |3|
|@DIOHz0r|30|

### References

Closes #564, Closes #676

### CODE REVIEW

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 1]
```diff
@@ -58,4 +58,29 @@ public function testGetTypeName() {
...
+    $instance = $this->createInstance();
+    list($policy) = $testPolicy->createNewPolicyInstance();
+        $status = $policy->filterStatus("pending");
+        $this->notNull($status);
+        $instance->update([
+           'id'     => $instance->getID(),
+           'status' => $status,
+        ]);
+        $this->string($status)->isEqualTo($instance->getField('status'));
+   }
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: Indentation problem ?

Comment (ajsb85) [reply tests/suite-unit/PluginFlyvemdmTaskstatus.php]: Hi, @flyve-mdm/backend-development 
@Gianfranco97 is using `.editorconfig` in the Web Dashboard repo.
Reference:
https://editorconfig.org

Comment (ingluife) [reply tests/suite-unit/PluginFlyvemdmTaskstatus.php]: ok, i will it.

Comment (btry) [reply tests/suite-unit/PluginFlyvemdmTaskstatus.php]: @ingluife have a look on this: https://glpi-developer-documentation.readthedocs.io/en/master/codingstandards.html

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 2]
```diff
@@ -35,6 +35,40 @@
...
+        'unicity'   => 0,
+     ];
+     /**
+      * @return array
+      */
+     private function createNewTaskInstance() {
+        $class = new \PluginFlyvemdmTask();
+        $task = $this->newTestedInstance($class);
+        return $task;
+     }
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: This method is useless. Replace calls to this method with $task = $this->newTestedInstance();

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 3]
```diff
@@ -35,6 +35,40 @@
...
+
+     public function beforeTestMethod($method) {
+        switch ($method) {
+            case 'testPrepareInputForUpdate':
+                parent::beforeTestMethod($method);
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: I would prefer to move this before the switch (){...}.

Also, if I remember correctly, no need to call yourself ```this->setupGLPIFramework();```

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 4]
```diff
@@ -35,6 +35,40 @@
...
+                $this->setupGLPIFramework();
+                $this->login('glpi', 'glpi');
+               break;
+         }
+     }
+     public function afterTestMethod($method) {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: This method is useless: tests runs in separate process as long they are parallelized; and before each test the framework is initialized, resetting the session.

**Comment by DIOHz0r** [comment]
Hello all, I've updated this PR with more tests please review it.

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 5]
```diff
@@ -35,12 +35,48 @@
...
+      $task = new \PluginFlyvemdmTask();
+      $task->add([
+         'value'                       => '0',
+         'plugin_flyvemdm_policies_id' => $policy->getID(),
+         'itemtype_applied'            => \PluginFlyvemdmFleet::class,
+         'items_id_applied'            => $fleet->getID(),
+         'itemtype'                    => '',
+         'items_id'                    => '',
+      ]);
+      return [$policy, $task, $fleet];
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: I would prefer a method to create a fleet (already exists), and an other to create a  task, given a fleet or an agent  as parameter.

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 6]
```diff
@@ -55,7 +91,153 @@ public function testClass() {
...
+      }
+   }
+
+   /**
+    * @tags testUpdateStatus
+    */
+   public function testUpdateStatus() {
+      $instance = $this->newTestedInstance();
+      $policy = new \PluginFlyvemdmPolicy();
+      $this->variable($instance->updateStatus(new \PluginFlyvemdmPolicyBoolean($policy),''))->isNull();
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: space after comma, or CS will fail

**Comment by btry** [comment]
@ajsb85 

image

there is ```style``` message type, but no ```cs```. https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits

**Comment by DIOHz0r** [comment]
> there is ```style``` message type, but no ```cs```. https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits

I fixed that @btry

**Comment by DIOHz0r** [comment]
This PR is ready to review!

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 7]
```diff
@@ -55,7 +70,164 @@ public function testClass() {
...
+   /**
+    * @tags testUpdateStatus
+    */
+   public function testUpdateStatus() {
+      $instance = $this->newTestedInstance();
+      $policy = new \PluginFlyvemdmPolicyBoolean(new \PluginFlyvemdmPolicy());
+      $this->variable($instance->updateStatus($policy, ''))->isNull();
+   }
+
+   public function displayTabForItemProvider() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: providerDisplayTabForItem()

File: tests/suite-unit/PluginFlyvemdmTaskstatus.php [hunk 8]
```diff
@@ -55,7 +70,164 @@ public function testClass() {
...
+   public function testDisplayTabContentForItem($item, $expected) {
+      $class = $this->testedClass->getClass();
+      ob_start();
+      $class::displayTabContentForItem($item);
+      $result = ob_get_contents();
+      ob_end_clean();
+      $this->string($result)->contains($expected);
+   }
+
+   public function tabNameForItemProvider() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmTaskstatus.php]: providerTabNameForItem()

**Comment by DIOHz0r** [comment]
@btry can you give again your review? Thx

**Pull request merged**